packages/core/src/parsers/linkProcessor.js

```diff
@@ -23,41 +41,141 @@ const defaultTagLinkMap = {
+  const relativeResourcePath = resourcePath.startsWith('/')
+    ? resourcePath.substring(1)
+    : resourcePath;
+  const relativeResourcePathWithNoExt = fsUtil.removeExtension(relativeResourcePath);
+  const isPageSrc = config.addressablePagesSource.includes(relativeResourcePathWithNoExt);
+  return isPageSrc;
+}
+
+function isValidFileAsset(resourcePath, config) {
+  const relativeResourcePath = resourcePath.startsWith('/')
+    ? resourcePath.substring(1)
+    : resourcePath;
+  const fileIgnore = ignore().add(config.ignore);
+  if (relativeResourcePath && fileIgnore.ignores(relativeResourcePath)) {
+    return true;
+  }
+  const fullResourcePath = path.join(config.rootPath, relativeResourcePath);
+  return utils.fileExists(fullResourcePath);
+}
+
+/**
+ * Serves as an internal intra-link validator. Checks if the intra-links are valid.
+ * If the intra-links are not suspected to not be valid, a warning message will be logged.
+ *
+ * @param {Object<any, any>} node from the dom traversal
+ * @param {string} cwf as flagged from {@link ComponentPreprocessor}
+ * @param {Object<any, any>} page config passed for page metadata access
+ * @returns {string} these string return values are for unit testing purposes only
+ */
+function validateIntraLink(node, cwf, config) {
+  // checks if the file indicated in the intra link exists
+  let resourcePath = getResourcePath(node);
+  if (!resourcePath || utils.isUrl(resourcePath) || resourcePath.startsWith('#')) {
+    return 'Not Intralink';
+  }
+
+  const err = `You might have an invalid intra-link! Ignore this warning if it was intended.
+'${resourcePath}' found in file '${cwf}'`;
+
+  if (resourcePath.startsWith(config.baseUrl)) {
+    // strip baseUrl from resourcePath
+    // keep the link absolute and only turn it into relative when needed
+    resourcePath = resourcePath.substring(config.baseUrl.length);
+  }
+
+  // remove hash portion (if any) in the resourcePath
+  const resourcePathUrl = url.parse(resourcePath);
+  if (resourcePathUrl.hash) {
+    resourcePath = resourcePathUrl.path;
+  }
+
+  if (resourcePath.endsWith('/')) {
+    // e.g. /userGuide/
+    // append index.html
+    const implicitResourcePath = `${resourcePath}index.html`;
+    if (!isValidPageSource(implicitResourcePath, config) && !isValidFileAsset(implicitResourcePath, config)) {
+      logger.warn(err);
+      return 'Intralink ending with "/" is neither a Page Source nor File Asset';
+    }
+    return 'Intralink ending with "/" is a valid Page Source or File Asset';
+  }
+
+  const hasNoFileExtension = path.posix.extname(resourcePath) === '';
+  if (hasNoFileExtension) {
+    // does not end with '/' and no file ext (e.g. /userGuide)
+    // check as file asset and implicit index.html
+    const implicitResourcePath = `${resourcePath}/index.html`;
+    const asFileAsset = resourcePath;
+    if (!isValidPageSource(implicitResourcePath, config) && !isValidFileAsset(asFileAsset, config)) {
```

Comment:
will need to check `implicitResourcePath` as an asset too

Reply:
Added this condition!